clients/cobol-lsp-vscode-extension/src/services/ControlFlowService.ts

```diff
@@ -103,19 +118,56 @@ class AnalysisTask {
 
 export class ControlFlowAnalysisService implements AnalysisServiceDelegate {
   private tasks: Map<string, AnalysisTask>;
-  private callbacks: Map<string, ControlFlowAnalysisCallback>;
+  private latestResults: Map<string, LatestResultData>;
   private diagnosticService: DiagnosticService;
 
   public constructor(
     private mainChannel?: vscode.OutputChannel,
     private logChannel?: vscode.LogOutputChannel,
   ) {
     this.tasks = new Map<string, AnalysisTask>();
-    this.callbacks = new Map<string, ControlFlowAnalysisCallback>();
     this.diagnosticService = new DiagnosticService();
+    this.latestResults = new Map<string, LatestResultData>();
+  }
+
+  public async invalidate(documentUri: string) {
+    const task = this.tasks.get(documentUri);
+    this.latestResults.delete(documentUri);
```

Comment:
Somebody is probably awaiting the promise we should resolve or reject it so that the caller does not get stuck forever.

Reply:
Fixed